jobs/resend_invitation_email/worker.go

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+package resend_invitation_email
+
+import (
+	"encoding/json"
+	"net/http"
+	"strconv"
+
+	"github.com/mattermost/mattermost-server/v5/app"
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/mattermost/mattermost-server/v5/shared/mlog"
+)
+
+const TwentyFourHoursInMillis int64 = 86400000
```

Comment:
Wondering if we can add this to the config or override with an env var for testing?

Reply:
We can change it to a ```var``` instead of ```const```. Though, I have spent the day trying to write a test but couldn't get it to work properly.

Issues:
- Need to have the CWS mocked out because of calls like this ```subscription, subErr := c.App.Cloud().GetSubscription(c.App.Session().UserId)```. This is not an issue since this [PR](https://github.com/mattermost/enterprise/pull/951) will provide the mock
- I somehow couldn't get the job to work in the test code. Even altering the polling time to a shorter time failed.

```go
func TestResendInvitationEmailJob(t *testing.T) {
	jobs.DefaultWatcherPollingInterval = 200

	th := api4.Setup(t).InitBasic()
	defer th.TearDown()

	user1 := th.GenerateTestEmail()
	user2 := th.GenerateTestEmail()

	emailList := []string{user1, user2}

	//Delete all the messages before check the sample email
	mail.DeleteMailBox(user1)
	mail.DeleteMailBox(user2)

	enableEmailInvitations := *th.App.Config().ServiceSettings.EnableEmailInvitations
	defer func() {
		th.App.UpdateConfig(func(cfg *model.Config) { cfg.ServiceSettings.EnableEmailInvitations = &enableEmailInvitations })
	}()

	th.App.UpdateConfig(func(cfg *model.Config) { *cfg.ServiceSettings.EnableEmailInvitations = true })

	checkEmail := func(t *testing.T, expectedSubject string) {
		//Check if the email was sent to the right email address
		for _, email := range emailList {
			var resultsMailbox mail.JSONMessageHeaderInbucket
			err := mail.RetryInbucket(10, func() error {
				var err error
				resultsMailbox, err = mail.GetMailBox(email)
				return err
			})
			if err != nil {
				t.Log(err)
				t.Log("No email was received, maybe due load on the server. Disabling this verification")
			}
			if err == nil && len(resultsMailbox) > 0 {
				require.True(t, strings.ContainsAny(resultsMailbox[len(resultsMailbox)-1].To[0], email), "Wrong To recipient")
				resultsEmail, err := mail.GetMessageFromMailbox(email, resultsMailbox[len(resultsMailbox)-1].ID)
				if err == nil {
					require.Equalf(t, resultsEmail.Subject, expectedSubject, "Wrong Subject, actual: %s, expected: %s", resultsEmail.Subject, expectedSubject)
				}
			}
		}
	}

	_, resp := th.SystemAdminClient.InviteUsersToTeamGracefully(th.BasicTeam.Id, emailList)
	api4.CheckNoError(t, resp)
	nameFormat := *th.App.Config().TeamSettings.TeammateNameDisplay
	expectedSubject := i18n.T("api.templates.invite_subject",
		map[string]interface{}{"SenderName": th.SystemAdminUser.GetDisplayName(nameFormat),
			"TeamDisplayName": th.BasicTeam.DisplayName,
			"SiteName":        th.App.ClientConfig()["SiteName"]})
	checkEmail(t, expectedSubject)

	//Delete all the messages before check whether the job resends emails
	mail.DeleteMailBox(user1)
	mail.DeleteMailBox(user2)

	time.Sleep(3 * time.Second)
	expectedSubject = i18n.T("api.templates.invite_subject",
		map[string]interface{}{"SenderName": "Noti",
			"TeamDisplayName": th.BasicTeam.DisplayName,
			"SiteName":        th.App.ClientConfig()["SiteName"]})
	checkEmail(t, expectedSubject)
}

```

Reply:
Yes, those changes will help for unit test - not sure if we have a testing framework for the jobs though. This would be something that we can consult with the platform team. 

However, if we are able to configure through env vars it can help with easily testing in our test environment

Reply:
@marianunez sorry i miss interpreted the comment. I thought you meant "tests". I'll change it to read from an env var.